packages/front-end/components/Settings/EditDataSource/DataSourceExperimentProperties/DataSourceEditExperimentEventPropertiesModal.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC } from "react";
+import { useForm } from "react-hook-form";
+import {
+  DataSourceEvents,
+  DataSourceInterfaceWithParams,
+} from "back-end/types/datasource";
+import Field from "../../../Forms/Field";
+import Modal from "../../../Modal";
+
+type DataSourceEditExperimentEventPropertiesProps = {
+  dataSource: DataSourceInterfaceWithParams;
+  onSave: (experimentProps: DataSourceEvents) => void;
+  onCancel: () => void;
+};
+
+export const DataSourceEditExperimentEventPropertiesModal: FC<DataSourceEditExperimentEventPropertiesProps> = ({
+  dataSource,
+  onCancel,
+  onSave,
+}) => {
+  const form = useForm<DataSourceEvents>({
+    defaultValues: {
+      experimentEvent: dataSource.settings?.events?.experimentEvent || "",
+      experimentIdProperty:
+        dataSource.settings?.events?.experimentIdProperty || "",
+      variationIdProperty:
+        dataSource.settings?.events?.variationIdProperty || "",
+    },
+  });
+
+  const handleSubmit = form.handleSubmit(async (value) => {
+    onSave(value);
+
+    form.reset({
+      experimentEvent: "",
+      experimentIdProperty: "",
+      variationIdProperty: "",
+    });
+  });
+
+  const dataSourceEvents = dataSource.settings.events;
+  if (!dataSourceEvents) {
+    console.error(
+      "ImplementationError: dataSource.settings.events cannot be null"
+    );
+    return null;
+  }
+
+  const saveEnabled = !!(
+    form.watch("experimentEvent") &&
+    form.watch("experimentIdProperty") &&
+    form.watch("variationIdProperty")
+  );
```

Comment:
This assumes all properties are required. Let me know if we want to change this validation logic.